### DISCUSSION TITLE
Translating homepage & translation related questions

### DISCUSSION THREAD

**Discussion by goshawk1980** [givanz/Vvveb#189] [General]
I've been away for a while a taking it back up where i left it.
I managed to get everything working but I can't figure out how to translate the homepage

The way i understand it works doesn't make much sense.

**admin/index.php?module=settings/site&site_id=1** 
Allows you to set a 'home template' but this is in the general tab, and can't be set by language
It should therefor be a lot easear if you could set the page you define as a homepage (as done in wordpress)

**admin/index.php?module=content/post&type=page**
Here you could then create a homepage and select a template for each language or create a homepage from scratch in every language

At the moment 'homepage' is only available in the template builder and editing will affect all languages.

I managed to install the language packs & can now edit those.
But this creates urls like fr_FR/page/portfolio-14
If i edit the translation code to fr instead, the url looks fine but i loose the vvveb translations in admin/index.php?module=localization/translations&action=domains&lang=fr so there should be a way to edit the url without it affecting the translation files, is there?

**Comment by givanz** [comment]
To translate homepage or any other page text  you can use the translate function in the page builder.

![translate](https://github.com/user-attachments/assets/1e574cd4-edf5-445e-9d88-7549518b5b40)

If you want to have different templates with different layout for each language you can activate `Language specific html template` plugin.

Then you can duplicate `index` template to `index.fr_FR` this works for other templates also, you can duplicate for example `/content/page` to `/content/page.fr_FR`  or `/contact.fr_FR.html` etc to load a different template based on selected language.

![template](https://github.com/user-attachments/assets/2a341333-6c07-49b2-872b-5e7b2ef62aaf)

The language code must match the language folder in `locale/fr_FR`, in order for translations to work with a shorter language code you must also change the folder name from `fr_FR` to `fr`. 

Homepage is rendered using active theme's index.html and it does not have a page associated like in WordPress, pages are only used to render `/page/{slug}` pages.

If you want to simulate WordPress behavior you can add a post component to homepage and choose the page you wish to show and the content will be changed according to selected language.

![post](https://github.com/user-attachments/assets/f6e0b91a-268a-431b-8e99-0675088574f3)

or just insert the html with code editor 

```html
<div data-v-component-post data-v-post_id="10">
  <h2 data-v-post-name>name</h2>
   <div data-v-post-content>content</div>
</div>
```

You will need to redownload `latest.zip` to get the most recent changes to language editor.

**Reply by goshawk1980** [reply]
Hi,

Thanks man! That's awesome :)

I will soon take the time to fully read the docs but I have to create a bunch of websites pretty fast and vvveb really does an amazing job at it! I thought i had hit a regrettable limitation but I'm glad to be wrong... the more I learn about vvveb, the more I like it :) 

I 100% prefer vvveb over wordpress to be honest... Powerful & elegant! Keep up the good work!

**Comment by givanz** [comment]
Thank you.

Let me know if you find any issue with translations.

**Reply by goshawk1980** [reply]
I started to debug the code cause I can't get it to work but...

When you say 'you should download the "latest"', you mean from the website right? ( at https://www.vvveb.com/ ) -or- should i download it from github -or- doesn't that matter at all? just to be sure I'm getting the right version to begin with :)

Because when i tried the component approach, i can't even find 'post', only 'posts' is showing up in the component section

**Reply by givanz** [reply]
The latest version with newest features can be downloaded from the website, the github code and release is updated later to allow more testing before push.

If post component is not showing you can try clearing browser cache to load the new js files.

**Reply by goshawk1980** [reply]
ok, so did a fresh install and everything seems to work using the template copy method. 
( Getting quiet some unrelated to this topic errors tho :( ... fe. when activating the SEO plugin )

**for info:** 
The reason it didn't work is because i tried to switch languages directly from the url but it appears a POST request is needed to change the site language on client side (using Language component fe.) ... 

**ps:** 
For some reason duplicating templates is not possible when the editor is on the current template. So when I am on the 'about us' page (client side) and click "design page" it is, in the editor, not possible to duplicate the 'about us' template at that point. Only the 'visit page' icon is available, not the trash bin or duplicate btn. Even after selecting another template in the same edit session, the 'about us' template can not be deleted or duplicated. I need to go back to the client side, re-enter the editor from fe. the homepage to duplicate the about-us page. :)

**Reply by goshawk1980** [reply]
was unable to delete index.fr_fr etc.. template files, was setting post_id to 'undefined'(string)

**admin/controller/editor.php :: r344**
//$post_id   = $this->request->post['post_id'] ?? false;
$post_id     = (isset($this->request->post['post_id']) && $this->request->post['post_id']!='undefined') ? $this->request->post['post_id'] : false;

**Reply by givanz** [reply]
Can you please provide the error messages from seo plugin?

I changed the code to work with get parameter also, now the language is changed when accessing https://demo.vvveb.com/fr_FR/  https://demo.vvveb.com/tr_TR/

I enabled duplicate/rename for currently edited page in the last update.

Thanks for the bug report, I fixed template delete issue, you will need to clear browser cache after update to load new js files.
